packages/help-center/src/components/help-center.tsx

```diff
@@ -2,51 +2,136 @@
 import type { HelpCenterSelect, SiteSelect, UserSelect } from '@automattic/data-stores';
 import '../styles.scss';
 
+const ZENDESK_SCRIPT_ID = 'ze-snippet';
+
 const HelpCenter: React.FC< Container > = ( { handleClose, hidden } ) => {
 	const portalParent = useRef( document.createElement( 'div' ) ).current;
 	const { data: chatStatus } = useSupportAvailability( 'CHAT' );
-	const { data } = useHappychatAvailable( Boolean( chatStatus?.is_user_eligible ) );
-	const { setShowHelpCenter } = useDispatch( HELP_CENTER_STORE );
-	const { setUnreadCount } = useDispatch( HELP_CENTER_STORE );
 	const { setSite } = useDispatch( HELP_CENTER_STORE );
+	const { setShowMessagingLauncher } = useDispatch( HELP_CENTER_STORE );
+	const { setShowMessagingWidget } = useDispatch( HELP_CENTER_STORE );
+	const { setThirdPartyCookiesAllowed } = useDispatch( HELP_CENTER_STORE );
 
-	const { show, isMinimized } = useSelect(
-		( select ) => ( {
-			isMinimized: ( select( HELP_CENTER_STORE ) as HelpCenterSelect ).getIsMinimized(),
-			show: ( select( HELP_CENTER_STORE ) as HelpCenterSelect ).isHelpCenterShown(),
-		} ),
-		[]
-	);
+	const { status: zendeskStatus } = useZendeskConfig( Boolean( chatStatus?.is_user_eligible ) );
+	useEffect( () => {
+		if ( zendeskStatus === 'success' ) {
+			setThirdPartyCookiesAllowed( true );
+		} else if ( zendeskStatus === 'error' ) {
+			setThirdPartyCookiesAllowed( false );
+		}
+	}, [ setThirdPartyCookiesAllowed, zendeskStatus ] );
+
+	useEffect( () => {
+		if ( ! chatStatus?.is_user_eligible ) {
+			return;
+		}
+
+		const zendeskKey: string | false = config( 'zendesk_support_chat_key' );
+		if ( ! zendeskKey ) {
+			return;
+		}
+
+		if ( document.getElementById( ZENDESK_SCRIPT_ID ) ) {
+			return;
+		}
+
+		function setUpMessagingEventHandlers() {
+			if ( typeof window.zE !== 'function' ) {
+				return;
+			}
+
+			window.zE( 'messenger', 'hide' );
+
+			window.zE( 'messenger:on', 'open', function () {
+				setShowMessagingWidget( true );
+			} );
+			window.zE( 'messenger:on', 'close', function () {
+				setShowMessagingWidget( false );
+			} );
+			window.zE( 'messenger:on', 'unreadMessages', function ( count ) {
+				if ( Number( count ) > 0 ) {
+					setShowMessagingLauncher( true );
+				}
+			} );
+		}
 
-	const { unreadCount, closeChat } = useHCWindowCommunicator( isMinimized || ! show );
+		loadScript(
+			'https://static.zdassets.com/ekr/snippet.js?key=' + encodeURIComponent( zendeskKey ),
+			setUpMessagingEventHandlers,
+			{ id: ZENDESK_SCRIPT_ID }
+		);
+	}, [ setShowMessagingLauncher, setShowMessagingWidget, chatStatus ] );
 
+	const { data: messagingAuth } = useMessagingAuth( Boolean( chatStatus?.is_user_eligible ) );
 	useEffect( () => {
-		setUnreadCount( unreadCount );
-	}, [ unreadCount, setUnreadCount ] );
+		const jwt = messagingAuth?.user.jwt;
+		if ( typeof window.zE !== 'function' || ! jwt ) {
+			return;
+		}
+
+		window.zE( 'messenger', 'loginUser', function ( callback ) {
+			callback( jwt );
+		} );
```

Comment:
Not sure how much of a good idea, but it may be nice to move this to `onSuccess` of `useMessagingAuth` query for fewer effects and  more cohesive code.

Reply:
Hmmm, I like the idea, but for now I'd prefer to keep it as-is. I'm worried about race conditions here and want to see how the code will perform in real scenarios, at WordPress.com's scale. The part I'm worried about is the `typeof window.zE !== 'function'` check. The Zendesk script needs to be loaded first, only then we can authenticate. In my testing it worked well, but I might need to adjust the code to be dependent on the `loadScript`'s callback to fire first.